include/GIL/Instructions/EnumVariantInst.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef GLU_GIL_INSTRUCTIONS_ENUM_VARIANT_INST_HPP
+#define GLU_GIL_INSTRUCTIONS_ENUM_VARIANT_INST_HPP
+
+#include "ConstantInst.hpp"
+#include "Member.hpp"
+
+namespace glu::gil {
+
+/// @class EnumVariantInst
+/// @brief Represents a specific enum variant as an instruction.
+///
+/// This instruction is used to refer to a specific variant of an enum type.
+/// It does not take any operands and simply holds a `Member` representing the
+/// variant.
+class EnumVariantInst : public ConstantInst {
+    Member member; ///< The enum variant represented as a Member.
+
+public:
+    /// @brief Constructs an EnumVariantInst from a Member.
+    ///
+    /// @param member The enum variant to represent.
+    EnumVariantInst(Member member)
+        : ConstantInst(InstKind::EnumVariantInstKind), member(member)
+    {
+        // return member.getType();
+    }
+
+    static bool classof(InstBase const *inst)
+    {
+        return inst->getKind() == InstKind::EnumVariantInstKind;
+    }
+
+    Operand getOperand(size_t index) const override
+    {
+        if (index == 0) {
+            return Operand(member);
+        }
+        llvm_unreachable("Invalid operand index");
```

Comment:
```suggestion
        assert(index == 0 && "Invalid operand index");
        return member;
```
it's simpler, clearer, and more predictable code gen.
in debug mode it's equivalent (it will crash for invalid input)
in release mode:
- with assert, it will simply return member no matter the given index, as the assert is just removed. 
- with llvm_unreachable, it will invoke UB if the index is not 0 (it will probably just remove the if and do the same as above)